erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py

```diff
@@ -543,9 +550,14 @@ def sync_shopify_orders():
 		create_order(order)
 
 def validate_customer_and_product(order):
-	if not frappe.db.get_value("Customer", {"shopify_id": order.get("customer").get("id")}, "name"):
-		create_customer(order.get("customer"))
-
+	customer_id = order.get("customer", {}).get("id")
+	if not customer_id:
+		if not frappe.db.get_value("Customer", {"shopify_id": customer_id}, "name"):
+			create_customer(order.get("customer"))
+	else:
+		frappe.msgprint(__("Customer is mandatory to create order"))
```

Comment:
This condition looks so strange.

The else part looks like `if customer_id` show message "Customer is mandatory". How does this work out?